llvm/lib/AsmParser/LLParser.cpp

```diff
@@ -3530,7 +3530,12 @@ bool LLParser::parseTargetExtType(Type *&Result) {
   if (parseToken(lltok::rparen, "expected ')' in target extension type"))
     return true;
 
-  Result = TargetExtType::get(Context, TypeName, TypeParams, IntParams);
+  auto TTy =
```

Comment:
Can we just use `targetexttype::get` here and just make `cantFail` check the error instead of checking here?

Reply:
I'm not sure what you're suggesting. The intention of this patch is that most callers of `TargetExtType::get` are unaffected. Only callers that want to opt in to error detection have to change to use the new `getOrError` instead.